### PR TITLE
Convert cast vote record optimization env vars to system settings

### LINKED ISSUE #4234: Switch from environment variables to system settings for the CAST_VOTE_RECORD_OPTIMIZATION_EXCLUDE_REDUNDANT_METADATA and CAST_VOTE_RECORD_OPTIMIZATION_EXCLUDE_ORIGINAL_SNAPSHOTS feature flags
https://github.com/usnistgov/CastVoteRecords/issues/45

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Closes #4234 

Switch the cast vote record optimization environment variables to system settings.

## Demo Video or Screenshot
N/A

## Testing Plan
Updated existing tests
## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: libs/types/src/system_settings.ts [hunk 1]
```diff
@@ -64,6 +64,16 @@ export interface SystemSettings {
...
+  /**
+   * Excludes original snapshots in cast vote record reports, decreasing export size and
+   * import/export time
+   */
+  readonly castVoteRecordOptimizationExcludeOriginalSnapshots?: boolean;
+  /**
+   * Excludes redundant metadata in cast vote record reports, decreasing export size and
+   * import/export time
+   */
+  readonly castVoteRecordOptimizationExcludeRedundantMetadata?: boolean;
```
Comment (arsalansufi) [inline libs/types/src/system_settings.ts:76]: Can we make it such that, if you don't include these fields in the `systemSettings.json`, we default to the behavior that we want in the overwhelming majority of cases (i.e. excluding this data)? (We really only want to include this data during cert and, even then, for just a tiny part of cert.)

Could do that by flipping these booleans, e.g., `castVoteRecordOptimizationIncludeOriginalSnapshots` and `castVoteRecordOptimizationIncludeRedundantMetadata`

**Review by arsalansufi — COMMENTED** [review]
One quick thought!

File: libs/types/src/system_settings.ts (hunk 1 above)
Comment (arsalansufi) [reply libs/types/src/system_settings.ts:76]: I guess `Optimization` doesn't really make sense in the names anymore if you flip haha

Comment (jonahkagan) [reply libs/types/src/system_settings.ts:76]: Would it be worth also just collapsing them into one flag, like `castVoteRecordStrictCdf` or something?

Comment (arsalansufi) [reply libs/types/src/system_settings.ts:76]: Yeah I like that!

Comment (arsalansufi) [reply libs/types/src/system_settings.ts:76]: Actually, on second thought, let's keep them separate. Just to have the flexibility. There may be cases where a customer wants original snapshots (for auditing purposes) but not the redundant metadata.

It's also not clear to me reading the CVR CDF spec whether or not original snapshots are required or optional.

File: libs/types/src/system_settings.ts [hunk 2]
```diff
@@ -65,15 +65,15 @@ export interface SystemSettings {
   readonly precinctScanDisallowCastingOvervotes: boolean;
   readonly precinctScanEnableShoeshineMode?: boolean;
   /**
-   * Excludes original snapshots in cast vote record reports, decreasing export size and
-   * import/export time
+   * Includes original snapshots in cast vote record reports, decreasing export size and
```
Comment (arsalansufi) [inline libs/types/src/system_settings.ts:68]: > ~decreasing~ increasing export size

I think you got this below but just missed here

**Review by arsalansufi — APPROVED** [review]
LGTM!

**Pull request merged**